README.md

```diff
@@ -102,10 +106,11 @@ dgraphClient.alter(Operation.newBuilder().setDropAll(true).build());
 
 ### Create a transaction
 
-To create a transaction, call `DgraphClient#newTransaction()` method, which returns a
-new `Transaction` object. This operation incurs no network overhead.
+There are two types of transactions in dgraph, i.e. the read-only transactions that only include queries and the transactions that change data in dgraph with mutate operations. Both the synchronous client `DgraphClient` and the async client `DgraphAsyncClient` support the two types of transactions by providing the `newTransaction` and the `newReadOnlyTransaction` APIs. Creating a transaction is a local operation and incurs no network overhead.
 
-It is good practise to call `Transaction#discard()` in a `finally` block after running
+If a transaction is _not_ read-only, it can have any number of query, or mutate operations, which means such a transaction may also include only query operations. However, _non_ read-only transactions place a heavier load on the dgraph cluster by requesting unique timestamps, while all read-only transactions can share the same timestamp. Therefore if a transaction contains only query operations, we strongly recommend to use a read-only transaction.
```

Comment:
Mostly people should be using the "normal" transaction. If you're only doing queries, then you might benefit from a read-only transaction, which can share the same read timestamp across multiple such read-only txns, thus, potentially providing better latency.